### PR TITLE
set up associations

### PR DESCRIPTION
**Pull request by nimplay**
- [x]  Make sure that there are no linter errors. 
- [x]  Make sure that you used correct Gitflow. 
- [x]  Make sure that you documented your work in a professional way
- [x] Create classroom file and Create a class Classroom
- [x] @label instance variable and Setter and getter for @label
- [x] create relationship between student classrom has-many, belongs-to
- [x] Create Book file and Create a class Book
- [x] @title and @author instance variables, Setters and getters for instance variables
- [x] Create rental file and Create a class Rental 
- [x] @date instance variable, Setter and getter for @date
- [x] create relationship between Rental and book
- [x] create relationship between Rental and person

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -8,4 +8,9 @@ def initialize(classroom, age, name = 'Unknown', parent_permission: true, id: ni
   def play_hooky
     'H¯\(ツ)/¯'
   end
+
+  def classroom=(classroom)
+    @classroom = classroom
+    classroom.student.push(self) unless classroom.student.include?(self)
+  end
```
Comment (sja-thedude) [inline student.rb:17]: - [x] Kindly make sure that add `attr_reader :classroom` before initialize method. This is required for setting the classroom for students.

File: person.rb [hunk 2]
```diff
@@ -20,6 +20,10 @@ def correct_name
     @name
   end
 
+  def add_rental(date, book)
+    @rentals.push(Rental.new(date, self, book))
+  end
+
```
Comment (sja-thedude) [inline person.rb:27]: - [x] Kindly make sure to add the getter and setter method for rentals.

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @nimplay, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ Professional Readme
✔️ Correct Gitflow
✔️ All linters passing

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class Rental
+  attr_accessor :date
+  attr_reader :person, :book
+
+  def initialize(book, person, date)
+    @person = person
+    @date = date
```
Comment (medaminedev66) [inline rental.rb:7]: [OPTIONAL] The `@date` instance variable should be initialized in the constructor.

**Review by medaminedev66 — APPROVED** [review]
Hi @nimplay 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 💯 
Congratulations! 🎉 

### Highlights
- Your code is clean.
- Your repo is professional.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**